src/Microsoft.DotNet.CodeAnalysis/PackageFiles/codeAnalysis.targets

```diff
@@ -7,4 +7,10 @@
     <AdditionalFiles Include="$(MSBuildProjectDirectory)\*.analyzerdata" />
     <AdditionalFiles Include="$(MSBuildProjectDirectory)\*.analyzerdata.$(Platform)" />
   </ItemGroup>
+
+  <PropertyGroup>
+  	<!-- We need this additional property because by default msbuild will populate $(CodeAnalysisRuleSet) to a non-empty value -->
+    <OverrideCodeAnalysisRuleSet Condition="'$(OverrideCodeAnalysisRuleSet)'==''">$(MSBuildThisFileDirectory)Microsoft.DotNet.CodeAnalysis.ruleset</OverrideCodeAnalysisRuleSet>
```

Comment:
I would like to avoid creating yet another property here if we can avoid it. I would much perfer we check for empty and only set it if it isn't already set.